emission/analysis/modelling/trip_model/greedy_similarity_binning.py

```diff
@@ -121,6 +121,7 @@ class label to apply:
         self.is_incremental = config['incremental_evaluation']
 
         self.bins: Dict[str, Dict] = {}
+        self.tripLabels=[]
```

Comment:
I don't see why we need a separate array here.
the bins already are indexed by the bin_id, and bin has the labels in it

```
"25": {
'feature_rows': ....
'labels':...
}
```

So I don't see what this is adding to the data structure

Reply:
The clustering.py file builds on `data_loc` dataframe that requires

    Trip 1   -->    bin No. (say 2)
    Trip 2.  -->    bin No. (say 3) 
    Trip 3   -->   bin No.( say 2)
    Trip 4   -->   bin No.( say 1)
    Trip 5   -->   bin No.( say 1)
    .
    .
    .

However, the  way they are stored here  is  .

```
  "1": { 'feature_rows': [ [feature of Trip4], [feature of Trip5]]
            'labels': ...
  }
  
  "2": { 'feature_rows': [ [feature of Trip1], [feature of Trip3]]
            'labels': ...
  }
  
  "3": { 'feature_rows': [ [feature of Trip2]]
            'labels': ...
  }
```

We can extract the trip and its bin from the way they are already stored, but creating a separate array initially itself feels more efficient

Reply:
But the problem with creating an additional data structure is that then we are changing production code to make analysis easier. We don't really need the trip -> bin mapping on production, and at some point, might want to include memoization as well. So I would prefer doing the additional work in `clustering.py` to create the dataframe from the output of the production model and not the other way around.

Reply:
That makes sense. I'll remove the extra data structure and move all the  additional computations to clustering.py.